src/detect-content.c

```diff
@@ -379,20 +379,26 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
     if (!(s->flags & SIG_FLAG_DSIZE)) {
         return TRUE;
     }
-
     int max_right_edge_i = SigParseGetMaxDsize(s);
     if (max_right_edge_i < 0) {
         return TRUE;
     }
 
     uint32_t max_right_edge = (uint32_t)max_right_edge_i;
+    int content_temp_length = 0;
 
     const SigMatch *sm = s->init_data->smlists[DETECT_SM_LIST_PMATCH];
     for ( ; sm != NULL; sm = sm->next) {
         if (sm->type != DETECT_CONTENT)
             continue;
         const DetectContentData *cd = (const DetectContentData *)sm->ctx;
         uint32_t right_edge = cd->content_len + cd->offset;
+
+        int distance_size = cd->distance;
+        int previous_content_length = content_temp_length;
+        content_temp_length = cd->content_len;
+        int content_total_length = previous_content_length + content_temp_length;
```

Comment:
This logic seems to be incorrect in some cases where no ```distance``` is used at all. ETPRO sid ```2803076``` is affected by this, even though it looks fine to me.

Example:
```
alert tcp any any -> any any (content:!"abcdefgh"; dsize:26; content:"abcdefghijklmnopqrstuvwxyz"; depth:26; sid:666; rev:1;)
```
Gives:
```
[13681] 24/7/2019 -- 11:06:54 - (detect-content.c:417) <Error> (DetectContentPMATCHValidateCallback) -- [ERRCODE: SC_ERR_INVALID_SIGNATURE(39)] - signature can't match as dsize 26, content 34, distance 0.
[13681] 24/7/2019 -- 11:06:54 - (detect-engine-loader.c:184) <Error> (DetectLoadSigFile) -- [ERRCODE: SC_ERR_INVALID_SIGNATURE(39)] - error parsing signature "alert tcp any any -> any any (content:!"abcdefgh"; dsize:26; content:"abcdefghijklmnopqrstuvwxyz"; depth:26; sid:666; rev:1;)" from file local.rules at line 539
```
As you can see the 2 contents are not related by distance/within, but still considered together.

It is a weird sig btw, because if one of the conditions match by definition the other can't match.

Reply:
taking a look.